src/journal/cat.c

```diff
@@ -151,8 +155,23 @@ static int run(int argc, char *argv[]) {
 
         if (argc <= optind)
                 (void) execl("/bin/cat", "/bin/cat", NULL);
-        else
+        else {
+                _cleanup_free_ char *s = NULL;
+                struct stat st;
+
+                if (fstat(has_errfd ? STDERR_FILENO : STDOUT_FILENO, &st) < 0)
+                        return log_error_errno(errno,
+                                               "Failed to fstat(%s): %m",
+                                               FORMAT_PROC_FD_PATH(has_errfd ? STDERR_FILENO : STDOUT_FILENO));
+
+                if (asprintf(&s, DEV_FMT ":" INO_FMT, st.st_dev, st.st_ino) < 0)
+                        return log_oom();
+
+                if (setenv("JOURNAL_STREAM", s, 1) < 0)
```

Comment:
hmm, as per more recent coding style, maybe write the third arg as `…, /* overwrite= */ true) < …`